docs/api/other-api/serve.md

```diff
@@ -0,0 +1,96 @@
+
+You can see the underlying `express` server configuration in [packages/react-router-serve/cli.ts][rr-serve-code]. By default, it uses the following Express middlewares (please refer to their documentation for default behaviors):
+
+- [`compression`][compression]
+- [`express.static`][express-static] (and thus [`serve-static`][serve-static])
+- [`morgan`][morgan]
+
+## `HOST` environment variable
+
+You can configure the hostname for your Express app via `process.env.HOST` and that value will be passed to the internal [`app.listen`][express-listen] method when starting the server.
+
+```shellscript nonumber
+HOST=127.0.0.1 npx react-router-serve build/index.js
+```
+
+```shellscript nonumber
+react-router-serve <server-build-path>
+# e.g.
+react-router-serve build/index.js
+```
+
+## `PORT` environment variable
+
+You can change the port of the server with an environment variable.
+
+```shellscript nonumber
+PORT=4000 npx react-router-serve build/index.js
+```
+
+## Development Environment
+
+Depending on `process.env.NODE_ENV`, the server will boot in development or production mode.
+
+The `server-build-path` needs to point to the `serverBuildPath` defined in `react-router.config.ts`.
+
+Because only the build artifacts (`build/`, `public/build/`) need to be deployed to production, the `react-router.config.ts` is not guaranteed to be available in production, so you need to tell React Router where your server build is with this option.
+
+In development, `react-router-serve` will ensure the latest code is run by purging the `require` cache for every request. This has some effects on your code you might need to be aware of:
+
+- Any values in the module scope will be "reset"
+
+  ```tsx lines=[1-3]
+  // this will be reset for every request because the module cache was
+  // cleared and this will be required brand new
+  const cache = new Map();
+
+  export async function loader({ params }) {
+    if (cache.has(params.foo)) {
+      return cache.get(params.foo);
+    }
+
+    const record = await fakeDb.stuff.find(params.foo);
+    cache.set(params.foo, record);
+    return record;
+  }
+
+  If you need a workaround for preserving cache in development, you can set up a [singleton][singleton] in your server.
+
+- Any **module side effects** will remain in place! This may cause problems but should probably be avoided anyway.
+
+  ```tsx lines=[3-6]
+  import { json } from "@react-router/node"; // or cloudflare/deno
+
+  // this starts running the moment the module is imported
+  setInterval(() => {
+    console.log(Date.now());
+  }, 1000);
+
+  export async function loader() {
+    // ...
```

Comment:
```suggestion
  ```tsx lines=[1-4]
  // this starts running the moment the module is imported
  setInterval(() => {
    console.log(Date.now());
  }, 1000);
```

Reply:
Not sure if that suggestion will apply correctly with the back ticks, but you see what I mean